avutil/avutil.go

```diff
@@ -980,6 +1009,110 @@ func (f *Frame) PacketDuration() int64 {
 	return int64(C.av_frame_get_pkt_duration(f.CAVFrame))
 }
 
+func (f *Frame) SideData(dataType FrameSideDataType) *FrameSideData {
+	cSideData := C.av_frame_get_side_data(f.CAVFrame, (C.enum_AVFrameSideDataType)(dataType))
+	if cSideData == nil {
+		return nil
+	}
+
+	return NewFrameSideDataFromC(unsafe.Pointer(cSideData))
+}
+
+type FrameSideData struct {
+	CFrameSideData *C.AVFrameSideData
+}
+
+func NewFrameSideDataFromC(cFrameSideData unsafe.Pointer) *FrameSideData {
+	return &FrameSideData{
+		CFrameSideData: (*C.AVFrameSideData)(cFrameSideData),
+	}
+}
+
+func (sd *FrameSideData) Size() int {
+	return int(sd.CFrameSideData.size)
+}
+
+func (sd *FrameSideData) Type() FrameSideDataType {
+	return FrameSideDataType(sd.CFrameSideData._type)
+}
+
+func (sd *FrameSideData) MotionVector(n int) *MotionVector {
+	mv := C.go_motion_vector(sd.CFrameSideData, C.int(n))
+	return NewMotionVectorFromC(unsafe.Pointer(&mv))
+}
+
+func (sd *FrameSideData) Metadata() *Dictionary {
+	if sd.CFrameSideData.metadata == nil {
+		return nil
+	}
+
+	return NewDictionaryFromC(unsafe.Pointer(&sd.CFrameSideData.metadata))
+}
+
+func (sd *FrameSideData) Free() {
+	if sd == nil {
+		return
+	}
+
+	C.av_buffer_unref(unsafe.Pointer(&sd.CFrameSideData.buf))
```

Comment:
We don't use `buf` directly here, but we need to free it.

I will dig more about the `buf` and `av_buffer_unref` usage.

Thanks.